File: src/main/java/de/steklopod/sec04/Lec08DefaultIfEmpty.java
package de.steklopod.sec04;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

public class Lec08DefaultIfEmpty {

    public static void main(String[] args) {

        getOrderNumbers()
                .filter(i -> i > 10)
                .defaultIfEmpty(-100)
                .subscribe(Util.subscriber());

    }

    private static Flux<Integer> getOrderNumbers(){
        return Flux.range(1, 12);
    }


}

File: src/main/kotlin/de/steklopod/sec04/Lec08DefaultIfEmpty.kt
package de.steklopod.sec04

import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux

object Lec08DefaultIfEmpty {
    @JvmStatic
    fun main(args: Array<String>) {
        orderNumbers
            .filter { i: Int -> i > 10 }
            .defaultIfEmpty(-100)
            .subscribe(subscriber())
    }

    private val orderNumbers: Flux<Int> = Flux.range(1, 12)
}
